src/makeCalculator.js

```diff
@@ -5,6 +5,32 @@
  */
 function makeCalculator() {
   // write code here
+  return {
+    result: 0,
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(number) {
+      this.result += number;
+    },
+    multiply(number) {
+      this.result *= number;
+    },
+    subtract(number) {
+      this.result -= number;
+    },
+    divide(number) {
+      this.result /= number;
+    },
+    operate(callback, number) {
+      callback.call(this, number);
```

Comment:
[CODE KNOWLEDGE] - When using the `operate` method, ensure that the `callback` is a function before calling it. This can be done using a check like `typeof callback === 'function'`. This will prevent runtime errors if a non-function is passed as a callback.